### ISSUE TITLE
Slack posts clearing after week has past

### ISSUE THREAD

**Issue by ThorntonMatthewD** [codeforgreenville/c4g-events#48]
![Screenshot_2023-11-19_22-27-13](https://github.com/hackgvl/slack-events-bot/assets/44626690/8e95cc6a-3dcb-4492-9acf-2637cbceef54)
![Screenshot_2023-11-19_22-27-22](https://github.com/hackgvl/slack-events-bot/assets/44626690/dfff75e8-edf6-4c0c-8ae0-659946bdfbc9)

I noticed further up in the [events channel](https://hackgreenville.slack.com/archives/C042530GC) that some of the event posts are showing up as empty. I'm not sure if this is intentional to try and keep the channel tidy, or if it isn't then and there is a discrepancy between events availability from the Events API and when the Slack bot stops checking for events for that week.

@bogdankharchenko's [PR](https://github.com/hackgvl/hackgreenville-com/pull/186) that migrates the Events API over to the HG.com Laravel app adds a bit of [negative buffer](https://github.com/hackgvl/hackgreenville-com/blob/ec49abebe7848f65d8bb4fbefffd323334f7ca3f/app-modules/event-importer/src/Services/EventBriteHandler.php#L77), so if this is a legitimate issue then it may self-resolve whenever the bot is set to pull from the new setup.

**Comment by oliviasculley** [comment]
This is a side effect of the slack bot continually keeping posts up to date I think, and the fact that events fall off from the list returned from the API eventually. However, there should be code to "freeze" posts once the week is up to keep the "final" week posts for posterity

**Comment by ThorntonMatthewD** [comment]
The new Events API implementation has `start_date` and `end_date` query params available, so maybe the bot can "laser in" on the current and next weeks by providing those values, and that way it won't be subject to events in the current week being pruned as the days roll on by.

**Comment by allella** [comment]
I don't think I purged the database, but we did to some reimports when the timezone situation was being investigated.

I have, in the past, purged the database and imported fresh to try and figure out bugs, but hopefully that's more rare with the Larvel work that's been done. 

Though, I wouldn't put it out of the realm of possibility that events beyond some past date would be at risk of going away.  I did mention to @bogdankharchenko that the Python API has a [concept of the maximum number of days in the past to fetch from Meetup and Eventbrite](https://github.com/hackgvl/hackgreenville-com/pull/186/files#r1394423139), and another configuration for how many days in the past to return if the consuming app doesn't set a start date.
